src/Mesh/DerivedData.C

```diff
@@ -802,6 +804,86 @@ genEsued( const std::vector< std::size_t >& inpoel,
   return std::make_pair( std::move(esued1), std::move(esued2) );
 }
 
+std::size_t
+genNbfacTet( std::size_t tnbfac,
+             const std::vector< std::size_t >& inpoel,
+             const std::vector< std::size_t >& t_triinpoel,
+             std::vector< std::size_t >& triinpoel )
+// *****************************************************************************
+//  Generate the number of boundary-faces on this chare/worker/mesh-partition.
+//  \warning This is for Tetrahedra only.
+//! \param[in] tnbfac Total number of boundary faces in the entire mesh.
+//! \param[in] inpoel Inteconnectivity of points and elements. These are the
+//!   node ids of each element of an unstructured mesh.
+//! \param[in] t_triinpoel Interconnectivity of points and boundary-face in the
+//!   entire mesh.
+//! \param[inout] triinpoel Interconnectivity of points and boundary-face in 
+//!   this mesh-partition.
+//! \return Number of boundary-faces on this chare/mesh-partition.
+//! \details This function generates the number of physical-boundaries and a 
+//!   vector with face-node connectivity for boundary-faces that are present 
+//!   in this mesh-partition.
+// *****************************************************************************
+{
+  std::size_t nbfac(0), nnpf(3);
+
+  if (tnbfac > 0)
+  {
+
+  Assert( !inpoel.empty(), "Attempt to call genNbfacTet() on empty container" );
+  Assert( !t_triinpoel.empty(), "Attempt to call genNbfacTet() on empty container" );
+  Assert( t_triinpoel.size()/nnpf == tnbfac, 
+                  "Incorrect size of triinpoel in genNbfacTet()" );
+
+  auto nptet = inpoel;
+  auto nptri = t_triinpoel;
+
+  // reduced inpoel and t_triinpoel
```

Comment:
"Reduced" in what sense? I think the the code itself, `unique`, is a good enough "documentation" here.

Reply:
Okay. Removed the comment.